service/storage_service.cc

```diff
@@ -907,7 +928,11 @@ future<> storage_service::handle_state_normal(inet_address endpoint) {
     // Order Matters, TM.updateHostID() should be called before TM.updateNormalToken(), (see CASSANDRA-4300).
     if (_gossiper.uses_host_id(endpoint)) {
         auto host_id = _gossiper.get_host_id(endpoint);
-        auto existing = tmptr->get_endpoint_for_host_id(host_id);
+        auto existing = get_owner_of_tokens(tmptr, tokens);
+        auto existing_host_id = existing ?
+            std::optional<locator::host_id>(_gossiper.get_host_id(*existing)) : std::nullopt;
```

Comment:
host_id is initialized to a null uuid by default, why do we need an optional?
the null uuid is invalid and will never be returned by get_host_id()

Reply:
As you see, the existing_host_id is used for logging only here, I want to avoid printing things like  00000000-0000 ...

Reply:
I'm not sure it's worth the complication.
Null host_id is pretty clear is not available